records/generator.go

```diff
@@ -27,9 +29,17 @@ type rrs map[string][]string
 // RecordGenerator contains DNS records and methods to access and manipulate
 // them. TODO(kozyraki): Refactor when discovery id is available.
 type RecordGenerator struct {
-	As       rrs
-	SRVs     rrs
-	SlaveIPs map[string]string
+	As         rrs
+	SRVs       rrs
+	SlaveIPs   map[string]string
+	httpClient http.Client
+}
+
+// NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
+func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
```

Comment:
We weren't reusing the HTTP client before. Is there any reason that was the case? I personally don't know, but, if you're confident it's safe to do it, I'll make the change to have it passed in.